tests/services/sessions/test_api.py

```diff
@@ -1,61 +1,307 @@
+    sessions = j(resp)
+    assert sessions == [newsession]
+
+    # Retrieve it
+    sid = newsession['id']
+    resp = await session_client.get(sid)
+    got = j(resp)
+    assert got == newsession
+    # Need to find a better solution to this.
+    await session_client.cleanup()
+
+async def test_delete(session_client):
+    resp = await session_client.create('foo/nb1.ipynb')
+    newsession = j(resp)
+    sid = newsession['id']
+
+    resp = await session_client.delete(sid)
+    assert resp.code == 204
+
+    resp = await session_client.list()
+    sessions = j(resp)
+    assert sessions == []
+
+    with pytest.raises(tornado.httpclient.HTTPClientError) as e:
+        await session_client.get(sid)
+    assert expected_http_error(e, 404)
+    # Need to find a better solution to this.
+    await session_client.cleanup()
+
+async def test_modify_path(session_client):
+    resp = await session_client.create('foo/nb1.ipynb')
+    newsession = j(resp)
+    sid = newsession['id']
+
+    resp = await session_client.modify_path(sid, 'nb2.ipynb')
+    changed = j(resp)
+    assert changed['id'] == sid
+    assert changed['path'] == 'nb2.ipynb'
+    # Need to find a better solution to this.
+    await session_client.cleanup()
+
+async def test_modify_path_deprecated(session_client):
+    resp = await session_client.create('foo/nb1.ipynb')
+    newsession = j(resp)
+    sid = newsession['id']
+
+    resp = await session_client.modify_path_deprecated(sid, 'nb2.ipynb')
+    changed = j(resp)
+    assert changed['id'] == sid
+    assert changed['notebook']['path'] == 'nb2.ipynb'
+    # Need to find a better solution to this.
+    await session_client.cleanup()
+
+async def test_modify_type(session_client):
+    resp = await session_client.create('foo/nb1.ipynb')
+    newsession = j(resp)
+    sid = newsession['id']
+
+    resp = await session_client.modify_type(sid, 'console')
+    changed = j(resp)
+    assert changed['id'] == sid
+    assert changed['type'] == 'console'
+    # Need to find a better solution to this.
+    await session_client.cleanup()
+
+async def test_modify_kernel_name(session_client, fetch):
+    resp = await session_client.create('foo/nb1.ipynb')
+    before = j(resp)
+    sid = before['id']
+
+    resp = await session_client.modify_kernel_name(sid, before['kernel']['name'])
+    after = j(resp)
+    assert after['id'] == sid
+    assert after['path'] == before['path']
+    assert after['type'] == before['type']
+    assert after['kernel']['id'] != before['kernel']['id']
+
+    # check kernel list, to be sure previous kernel was cleaned up
+    resp = await fetch('api/kernels', method='GET')
+    kernel_list = j(resp)
+    after['kernel'].pop('last_activity')
+    [ k.pop('last_activity') for k in kernel_list ]
+    assert kernel_list == [after['kernel']]
+    # Need to find a better solution to this.
+    await session_client.cleanup()
+
+#### NOT FINISHED WITH THIS TEST
+async def test_modify_kernel_id(session_client, fetch):
```

Comment:
In comparing this test with the previous, it seems to be doing what the previous did.  Are there other thing you wanted to add?

Reply:
Oops. That’s an outdated comment I left for myself, because I got interrupted between flights 😂. I’ll remove that line.